README.md

```diff
@@ -473,22 +485,193 @@ When the scanner successfully decodes a `TransactionRequest` it will call its de
+```
+
+Where:
+* `apiKey` is the API key (typically generated on the admin panel).
+* `authenticationToken` is the token corresponding to an OmiseGO Wallet user retrievable using one of our server-side SDKs.
+> You can find more info on how to retrieve this token in the [OmiseGO server SDK documentations](https://github.com/omisego/ruby-sdk#login). 
+* `baseURL` is the URL of the OmiseGO Wallet API, this needs to be an ws(s) url. **Note**: This need to be ended with '/'.
+* `debug` is a boolean indicating if the SDK should print logs in the console.
+
+## Listen for the system event
+
+The system event is giving general information related to the status of the web socket connection.
+All listenable system events are the **Connection status** and the **Channel status**. 
+
+### Connection Event
+
+`SocketConnectionCallback` *(optional)* is the callback that listens for a **web socket's server connection status**. The possible events are:
+
+* `onConnected()`: Invoked when the web socket client has connected to the eWallet web socket API successfully.
+* `onDisconnected(throwable: Throwable)`: Invoked when the web socket client has disconnected from the eWallet web socket API.
+Throws an exception if the web socket was not disconnected successfully.
+
+**Usage**
+```kotlin
+socketClient.setConnectionListener(object : SocketConnectionCallback {
+    override fun onConnected() {
+        // Do something
+    }
+
+    override fun onDisconnected(throwable: Throwable?) {
+        // Do something
+    }
+})
+```
+### Channel Event
+
+`SocketChannelCallback` *(optional)* is the callback that listens for a **channel connection status**. The possible events are:
+
+* `onJoinedChannel(topic: SocketTopic)`: Invoked when the client have been joined the channel successfully.
+* `onLeftChannel(topic: SocketTopic)`: Invoked when the client have been left the channel successfully.
+* `onError(apiError: APIError)`: Invoked when something goes wrong while connecting to a channel.
+    
+**Usage**
+```kotlin
+socketClient.setChannelListener(object : SocketChannelCallback {
+    override fun onJoinedChannel(topic: SocketTopic) {
+        // Do something
+    }
+
+    override fun onLeftChannel(topic: SocketTopic) {
+        // Do something
+    }
+
+    override fun onError(apiError: APIError) {
+        // Handle an error
+    }
+})
+```
+
+## Listen for the custom event
+
+The custom event is the special event that currently limited to the `TransactionRequest` and the `TransactionConsumption` event.
+All custom events will be a sub-class of the `SocketCustomEventCallback`. `SocketCustomEventCallback` is the **required** generic callback that you will need to pass its sub-class when joining to the channel for listening for the event.
+All possible callbacks are the following:
+
+### TransactionRequest Event
+
+When creating a `TransactionRequest` that requires a confirmation it is possible to listen for all incoming using the `TransactionRequestCallback`.
```

Comment:
for all incoming what? confirmations? should probably precise :)

Reply:
Fixed